bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -0,0 +1,275 @@
+    } else
+    {
+        error("GetNetworkParams failed\n");
+    }
+
+    //
+    // Enumerate all of the adapter specific information using the IP_ADAPTER_INFO structure.
+    // Note:  IP_ADAPTER_INFO contains a linked list of adapter entries.
+    //
+    AdapterInfoSize = 0;
+	Err = GetAdaptersInfo(NULL, &AdapterInfoSize);
+    if (Err != 0)
+    {
+        if (Err != ERROR_BUFFER_OVERFLOW)
+            error("GetAdaptersInfo sizing failed\n");
+    }
+
+    // Allocate memory from sizing information
+	pAdapterInfo = (PIP_ADAPTER_INFO)malloc(AdapterInfoSize);
+    if (pAdapterInfo == NULL)
+        error("Memory allocation error\n");
+
+	char** IPs = malloc(AdapterInfoSize);
+    // Get actual adapter information
+	Err = GetAdaptersInfo(pAdapterInfo, &AdapterInfoSize);
+    if (Err != 0)
+        error("GetAdaptersInfo failed\n");
+
+    pAdapt = pAdapterInfo;
+
+    while (pAdapt)
+    {
+        pAddrStr = &(pAdapt->IpAddressList);
+        while(pAddrStr)
+        {
+            if(strcmp(pAddrStr->IpAddress.String, "0.0.0.0")){
+                printf("\tIP Address. . . . . . . . . : %s\n", pAddrStr->IpAddress.String);
+                IPs[j] = pAddrStr->IpAddress.String;
+                j += 1;
+            }
+            pAddrStr = pAddrStr->Next;
+        }
+
+        pAdapt = pAdapt->Next;
+    }
+    *addrCount = j;
+    return IPs;
+}
+
+int sendRequest(wchar_t* server, wchar_t* tunnel, wchar_t* reqData){
+    const wchar_t page[] = L"/windows";
+	const wchar_t requestType[] = L"POST";
+    HINTERNET hInternet, hConnect, hRequest;
+    wprintf(L"%ls : %ls : %ls\n", server, tunnel, reqData);
+    int finished = 0;
+    printf("1\n");
+    if (tunnel != NULL){
+        hInternet = InternetOpen(L"Mozilla/5.0", INTERNET_OPEN_TYPE_PROXY, tunnel, NULL, 0);
+    } else {
+        hInternet = InternetOpen(L"Mozilla/5.0", INTERNET_OPEN_TYPE_DIRECT, NULL, NULL, 0);
+    }
+    printf("2\n");
+    if (hInternet == NULL) {
+        printf("InternetOpen error : <%lu>\n", GetLastError());
+        return 1;
+    }
+    printf("3\n");
+    hConnect = InternetConnect(hInternet, server, BOOTLOADER_SERVER_PORT, L"", L"", INTERNET_SERVICE_HTTP, 0, 0);
+    if (hConnect == NULL) {
+        printf("hConnect error : <%lu>\n", GetLastError());
+        return 1;
+    }
+    printf("4\n");
+    hRequest = HttpOpenRequest(hConnect, requestType, page, NULL, NULL, NULL, 0, 0);
+    if (hRequest == NULL) {
+        printf("hRequest error : <%lu>\n", GetLastError());
+        return 1;
+    }
+    printf("5\n");
+    BOOL isSend = HttpSendRequest(hRequest, NULL, 0, reqData, (DWORD)(sizeof(wchar_t) * wcslen(reqData)));
+    if (!isSend){
+        printf("HttpSendRequest error : (%lu)\n", GetLastError());
+        // close request
+        InternetCloseHandle(hRequest);
+        InternetCloseHandle(hConnect);
+        InternetCloseHandle(hInternet);
+        return 1;
+    }
+    DWORD dwFileSize;
+	dwFileSize = BUFSIZ;
```

Comment:
Spacing

Reply:
On my IDE 4 spaces and a tab appeared to be the same size